protos/core.proto

```diff
@@ -424,4 +429,17 @@ message TableMetadata {
   Target target = 1;
   string type = 2; // "table" or "view"
   repeated Field fields = 3;
+  int64 last_updated_millis = 4;
+}
+
+message PersistedTableDependency{
+  Target target = 1;
+  int64 last_updated_millis = 2;
+}
+
+message PersistedTableMetadata {
+  Target target = 1;
+  string create_query_hash = 2;
```

Comment:
nit: I think this is a bit misleading, as it's more than just the create query, and for other files in the future (e.g.operatations) there isn't necessarily a create query at all.

I'd suggest something like: definition_hash or just hash, that's generic?